### PR TITLE
Fixes for HPC

### PR DESCRIPTION
**Pull request by zfergus**
Testing some fixes for needed HPC on GitHub Actions

### CODE REVIEW

File: cmake/recipes/mkl.cmake [hunk 1]
```diff
@@ -400,6 +400,10 @@ if(NOT MSVC)
     target_link_libraries(mkl::mkl INTERFACE ${LIBM_LIBRARY})
 endif()
 
+if(NOT MSVC)
+    target_link_libraries(mkl::mkl INTERFACE ${CMAKE_DL_LIBS})
```
Comment (jdumas) [inline cmake/recipes/mkl.cmake:404]: Why is this needed specifically on the HPC?

Comment (zfergus) [reply cmake/recipes/mkl.cmake:404]: Idk exactly, but if I don't have it linker errors:

```
/usr/bin/ld: ../_deps/mkl-static-src/lib/libmkl_core.a(mkl_memory_patched.o): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
//usr/lib64/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

I am using `gcc (GCC) 10.2.0`.

Comment (zfergus) [reply cmake/recipes/mkl.cmake:404]: I don't get this linker error on any of our other Linux machines.

Comment (zfergus) [reply cmake/recipes/mkl.cmake:404]: I also get a lot of linker errors from MKL when building TBB statically (and using the TBB backend for MKL).

```
../_deps/mkl-static-src/lib/libmkl_tbb_thread.a(zpotrf_tbb.o): In function `mkl_lapack_zpotrf':
zpotrf_tbb.cpp:(.text+0x3fe): undefined reference to `tbb::detail::r1::initialize(tbb::detail::d1::task_group_context&)'
zpotrf_tbb.cpp:(.text+0x41a): undefined reference to `tbb::detail::r1::notify_waiters(unsigned long)'
zpotrf_tbb.cpp:(.text+0x42c): undefined reference to `tbb::detail::r1::execute_and_wait(tbb::detail::d1::task&, tbb::detail::d1::task_group_context&, tbb::detail::d1::wait_context&, tbb::detail::d1::task_group_context&)'
zpotrf_tbb.cpp:(.text+0x435): undefined reference to `tbb::detail::r1::destroy(tbb::detail::d1::task_group_context&)'
```

@jdumas, have you encountered this issue before?

Comment (jdumas) [reply cmake/recipes/mkl.cmake:404]: Not sure what's causing the dlclose issue on some Linux flavor and not others (maybe an old version of glibc?). For the TBB issue, I think we've fixed it by updating our TBB version to v2021.6.0-rc1 (you can copy the recipe from Lagrange again).

**Comment by zfergus** [comment]
Thanks @jdumas. I updated TBB and now it compiles with TBB built statically. The only downside is SuiteSparse CMake complains about the newer TBB version:
```
CMake Warning at build/release/_deps/suitesparse-src/CMakeLists.txt:127 (find_package):
  Could not find a configuration file for package "TBB" that is compatible
  with requested version range "4.2...<2021.4".

  The following configuration files were considered but not accepted:

    /home/zjf214/scratch/polysolve/build/release/TBB/TBBConfig.cmake, version: 2021.6.0
```
but it seems like this is only used to make SPQR parallel:
```
-- The following RECOMMENDED packages have not been found:

 * TBB (required version 4.2...<2021.4), C++ template library for task parallelism, <https://www.threadingbuildingblocks.org>
   Enables multithreading support for SPQR
```

**Comment by jdumas** [comment]
Yeah, I guess you'll need to make a PR on the SuiteSparse repo ... I don't know why they would impose a maximum version on TBB ¯\\_(ツ)_/¯
